compiler/rustc_hir/src/lang_items.rs

```diff
@@ -266,7 +266,9 @@ language_item_table! {
     EhPersonality,           sym::eh_personality,      eh_personality,             Target::Fn,             GenericRequirement::None;
     EhCatchTypeinfo,         sym::eh_catch_typeinfo,   eh_catch_typeinfo,          Target::Static,         GenericRequirement::None;
 
-    OwnedBox,                sym::owned_box,           owned_box,                  Target::Struct,         GenericRequirement::Minimum(1);
```

Comment:
We already had parts of the compiler assume that there are two generic arguments.

I guess we can make it robust to having a single argument if needed, but it doesn't seem worth it. When is `Minimum` ever a good idea?

Reply:
I wrote this originally, and during my testing at the time, it was possible to write a `Box<T>` declaration without an allocator generic.  It's quite possible that I didn't exercise enough of the compiler, or that things changed in the meantime.  Additionally I tried to be conservative with these requirements when I was not sure, this is a best-effort check to try and catch the most obvious cases that people writing certain `no_std` binaries (or `no_core` for various reasons) encounter.

Reply:
Yeah it's possible to write them but parts of the compiler will ICE. This PR now fixes the ICE candidate I know about but I assume most people will assume that the number of generic args matches what std does, so any other number has a pretty high risk of ICEs.